src/yunohost/domain.py

```diff
@@ -218,3 +237,57 @@ def domain_remove(auth, domain, force=False):
     hook_callback('post_domain_remove', args=[domain])
 
     msignals.display(m18n.n('domain_deleted'), 'success')
+
+
+def domain_letsencrypt(auth, domain, create=False, renew=False, revoke=False):
+    """
+    Manage let's encrypt certificate for a domain
+
+    Keyword argument:
+        domain -- Domain to delete
+        create -- Create a Let's encrypt certificate
+        renew -- Force renewal of a Let's encrypt certificate
+        revoke -- Revoke a Let's encrypt certificate
+
+    """
+    if domain not in domain_list(auth)['domains']:
+        raise MoulinetteError(errno.EINVAL, m18n.n('domain_unknown'))
+
+    if create and not renew and not revoke:
+
+        # backup self signed certificate if exist
+        if os.path.exists('/etc/yunohost/certs/%s/cert.pem' % domain):
+            os.system('mkdir -p /etc/yunohost/certs/%s/yunohost_self_signed' % domain)
```

Comment:
Actually, you should use os.makedirs for this case https://docs.python.org/2/library/os.html#os.makedirs